pyibisami/ami_parse.py

```diff
@@ -185,22 +185,13 @@ def input_ami_param(self, params, pname):
 def lexeme(p):
     """Lexer for words."""
     return p << ignore  # skip all ignored characters.
-
-def int2tap(x):
-    """Convert integer to tap position."""
-    if (x[0] == '-'):
-        res = ("pre" + x[1:])
-    else:
-        res = ("post" + x)
-    return res
-
-
+    
 lparen = lexeme(string("("))
 rparen = lexeme(string(")"))
 number = lexeme(regex(r"[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?"))
 integ  = lexeme(regex(r"[-+]?[0-9]+"))
 nat    = lexeme(regex(r"[0-9]+"))
-tap_ix = integ.parsecmap(int2tap)
```

Comment:
The "pre" and "post" are necessary, in order to prevent the _Traits/UI_ machinery from barfing when it attempts to create the parameter adjustment dialog. That machinery doesn't seem to like attribute names beginning with numerals, let alone minus signs. :(

Reply:
Got you!

However the Matlab generated .so file does not accept the pre/post strings, but needs the -1,0,1 syntax. I'm not sure if this is according to spec, I thought the Model parameters should be named exactly like in the .ami file.

Reply:
Hi Franz,

Apologies for the long delay in responding; I'm not sure how I missed this. :(

So, it sounds like we've got 3 orthogonal issues tangled up here:

1. I can't give AMI parameters names that begin w/ a numeral, due to limitations in the `Traits/UI` package, upon which this entire project depends.
2. MATLAB insists on naming these parameters: "-1", "0", etc. (Do you have any control over this?)
3. Should model parameters sent to the DLL/SO have names matching the AMI file exactly? (Yes, they must.)

Is that a complete answer?